homeassistant/components/huawei_lte/sensor.py

```diff
@@ -351,7 +351,10 @@ class HuaweiSensorEntityDescription(SensorEntityDescription):
             "txpower": HuaweiSensorEntityDescription(
                 key="txpower",
                 name="Transmit power",
-                device_class=SensorDeviceClass.SIGNAL_STRENGTH,
+                # This is state_class=SensorDeviceClass.SIGNAL_STRENGTH by nature,
+                # but the value tends to consist of several, e.g.
+                #     PPusch:15dBm PPucch:2dBm PSrs:42dBm PPrach:1dBm
```

Comment:
If that is the case, I would say this sensor isn't correct, to begin with?
It shouldn't stuff multiple sensor values into a single sensor state.

Reply:
"If that is the case, I would say this sensor isn't correct, to begin with?"

It was marked as a question, as well... I don't know :)
I'm not sure what the values mean or how they can/should be used.

The thing I do know is that (in general) putting multiple values in a single state makes them less useful (or at least harder to use in e.g., automations and other places).

Another fourth option could be checking if it is a numeric value? (In that case, use the device class).

I'm also fine with just applying this change, but it did make me question the usefulness of the sensor provided.

Reply:
Agree wholeheartedly that a composite state like this is not good. But I'd rather not go as far as remove the sensor just for this reason as it's been there since I don't know when.

Your fourth option sounds like the best of the bunch to me, in a way. Did that in 44e95f8ea9331dc96f72670927f07f1ca609493b, with an additional sanity check on the unit. Didn't find a way to do that apart from using a deprecated import, and hit an infinite loop if trying to use state/unit_of_measurement (-> used the native ones). I'm feeling this is a bit overengineered for such a corner case purpose. Would personally just drop that change and go with the original, but can live with this one too if you think it's preferable.

Reply:
`DEVICE_CLASS_UNITS` isn't a public API of sensor (it isn't re-exported either); it should not be used.

Reply:
> still leaving up to you to choose between the two evils :)

Lol wow, thanks! 🙄 🤣